src/main/java/com/google/jenkins/plugins/computeengine/ComputeEngineInstance.java

```diff
@@ -84,21 +90,65 @@ protected void _terminate(TaskListener listener) throws IOException, Interrupted
         try {
             ComputeEngineCloud cloud = getCloud();
 
-            Computer computer = this.toComputer();
-            if (this.oneShot && this.createSnapshot && computer != null && !computer.getBuilds().failureOnly().isEmpty()) {
-                LOGGER.log(Level.INFO, "Creating snapshot for node ... " + this.getNodeName());
-                cloud.getClient().createSnapshot(cloud.projectId, this.zone, this.getNodeName());
-            }
+            ExecutorService executor = Executors.newFixedThreadPool(1);
```

Comment:
I would use `Computer.threadPoolForRemoting`

Reply:
Agreed, creating a new thread pool for each individual terminate call is inefficient.